src/event_field_transformations.cpp

```diff
@@ -17,6 +25,29 @@ static std::vector<cata_variant> species_of_monster( const cata_variant &v )
     return result;
 }
 
+static std::vector<cata_variant> terrain_flag( const cata_variant &v )
```

Comment:
I would name this `flags_of_terrain`.  Then when it's used like
```
"new_fields": { "terrain_flag": { "flags_of_terrain": "terrain" } }
```
the concepts are near each other visually: "flag" near "flag" and "terrain" near "terrain".